react/client/src/components/formPages/Step1.tsx

```diff
@@ -1,57 +1,159 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import Textarea from 'components/Textarea';
 import Input from 'components/Input';
 import Button from 'components/Button';
 import TextPreview from 'components/TextPreview';
+import RadioGroup from 'components/RadioGroup';
+import PopUp from 'components/PopUp';
+
+import 'styles/Step1.css';
 
 const Step1 = ({ inputs, setInputs, goToPage }: StepProps) => {
+  const [step1Inputs, setStep1Inputs] = useState({
+    fullName: '',
+    age: '',
+    isVeteran: '',
+  });
+  const [step, setStep] = useState('name');
+
   const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const inputName = e.currentTarget.name;
     const inputValue = e.currentTarget.value;
+
+    // Update Form state (old)
     if (inputName === 'name') {
-      setInputs({ ...inputs, name: inputValue });
+      setStep1Inputs({ ...step1Inputs, fullName: inputValue });
     } else if (inputName === 'age') {
-      setInputs({ ...inputs, age: inputValue });
-    } else if (inputName === 'introduction') {
-      setInputs({ ...inputs, introduction: inputValue });
+      setStep1Inputs({ ...step1Inputs, age: inputValue });
+    } else if (inputName === 'isVeteran') {
+      setStep1Inputs({ ...step1Inputs, isVeteran: inputValue });
     }
   };
 
+  useEffect(() => {
+    const formDiv: any = document.querySelector('.div-form');
+    const ageDiv: any = document.querySelector('.div-age');
+    const veteranDiv: any = document.querySelector('.div-veteran');
+    const popUp: any = document.querySelector('.div-popUp');
+    const preview: any = document.querySelector('.div-preview');
+    const nextButton: any = document.querySelector('.div-nextButton');
+    const looksGoodButton: any = document.querySelector('.div-looksGoodButton');
+
+    if (step1Inputs.fullName !== '') {
+      setStep('age');
+    }
+    if (step1Inputs.age !== '') {
+      setStep('isVeteran');
+    }
+    if (step1Inputs.isVeteran !== '') {
+      setStep('beforePreview');
+    }
+    if (step === 'preview') setStep('preview');
+
+    switch (step) {
+      case 'name':
+        formDiv.style.display = 'block';
+        preview.style.display = 'none';
+        looksGoodButton.style.visibility = 'hidden';
+        break;
+      case 'age':
+        ageDiv.style.visibility = 'visible';
+        ageDiv.firstChild.style.color = 'black';
+        break;
+      case 'isVeteran':
+        veteranDiv.style.visibility = 'visible';
+        veteranDiv.firstChild.style.color = 'black';
+        break;
+      case 'beforePreview':
+        popUp.style.visibility = 'hidden';
+        nextButton.style.visibility = 'visible';
+        break;
+      case 'preview':
+        formDiv.style.display = 'none';
+        nextButton.style.visibility = 'hidden';
+        preview.style.display = 'block';
+        looksGoodButton.style.visibility = 'visible';
```

Comment:
The design is interesting that the fields are hidden. Directly altering dom is anti-pattern to having a shadow dom, but what you could hide it.

```
 { step !== 'name' &&
         <Textarea
              inputName="name"
              placeholder="FirstName LastName"
              handleChange={handleChange}
              multi={false}
          />
 }
``` 
This isn't quite right since it'll also hide when you go past the step. Maybe step can be numerical and you can see `step > 1`, etc.